### ISSUE TITLE
[Linux] support flutter sdk distribution in the form of RPM and DEB packages

### ISSUE THREAD

**Issue by esdnm** [flutter/flutter#108681]
This are the two most common packaging format in the Linux world, please provide them.

**Comment by maheshmnj** [comment]
Hi @esdnm, Thanks for filing the issue. If i understand it right you are proposing  .deb  and  .rpm for build distribution is that correct?

**Comment by esdnm** [comment]
> Hi @esdnm, Thanks for filing the issue. If i understand it right you are proposing .deb and .rpm for build distribution is that correct?

ohh, that would be a separate feature request on its own.

In this issue I was asking for flutter itself.

**Comment by maheshmnj** [comment]
> ohh, that would be a separate feature request on its own.

> In this issue I was asking for flutter itself.

@esdnm Do you mean the flutter packages on pub should be provided as RPM and Deb packages?

**Comment by esdnm** [comment]
> > ohh, that would be a separate feature request on its own.
> 
> > In this issue I was asking for flutter itself.
> 
> @esdnm Do you mean the flutter packages on pub should be provided as RPM and Deb packages?

No, I am asking for flutter, the software that this git repo is providing.

**Comment by maheshmnj** [comment]
Thanks for the clarification, I am quite not sure if this is possible, but I am leaving this proposal for further insights from the team.

cc: @stuartmorgan

**Comment by suwermave** [comment]
Please no, add flatpak support instead ! 
It will be universal for all linux distrubutions! #41737
And drop Snap, it's the opposite of universal!

**Comment by KurtB03** [comment]
> Please no, add flatpak support instead ! It will be universal for all linux distrubutions! #41737 And drop Snap, it's the opposite of universal!

I don't think flatpak is really suitable for this kind of software (yet). And many people for many reasons could benefit from binary Packages. But if you really want it, it is not exclusive, open your PR to said issue

**Comment by esdnm** [comment]
hi @maheshmnj, any progress on this issue?

**Comment by maheshmnj** [comment]
@esdnm Unfortunately no updates so far, If there are any updates you will see them on this issue thread. This issue is mainly a new feature request and it is not yet ready to be worked on unless this issue is prioritized.

**Comment by esdnm** [comment]
hey @maheshmnj, what is holding this from resolving? Please make this happen!

**Comment by maheshmnj** [comment]
@esdnm Perhaps the flutter team is busy with some other important issues at this time, Should they think that this issue is affecting a lot of users you should eventually see some updates on this issue.

**Comment by maheshmnj** [comment]
People who are affected by this issue please leave an upvote on the [original issue report](https://github.com/flutter/flutter/issues/108681#issue-1323268517) it is one among many factors to prioritize the issue.

**Comment by SchoolGuy** [comment]
I asked for this in September 2020 already. I think this is important to make it easier for developers in a corporate environment to use it. Building such a toolchain from source is very inconvenient for people since the environments differ much more than in a centralized build chain of the OS vendor that builds a package for you.

**Comment by Karlosbubi** [comment]
I presume for Corporate support isn't *that* important, and for now sufficiently covered by the snap.
As indicated by the triage-lable the proposal was generally accepted but of no priority. 
It might be accelerated by a PR, but i don't know if the issues around the access has been solved, which are a likely blockers for "normal" linux installation.

**Comment by SchoolGuy** [comment]
Hm while I understand your reasoning I think snap is not appropriate in this scenario since it requires a special daemon that a small subset of systems has available. Flatpack has a larger market share but in general is due to the same concerns potentially problematic.

An Appimage would be an improvement if we must use a container like tool. I would gladly not have RPM/DEB packages but I see no way around it. If something is a DEB/RPM one can easily build all the other fancy new container formats derived from that.

Sadly I have no idea what business incentive could be created to make this a feature that you yourself desire. I am in a position where I can help from a packaging perspective but the biggest problem is the requirement for internet as most packaging is done in airgapped environments...

**Comment by Karlosbubi** [comment]
I don't think container like is the issue. I personally think AppImage and Flatpaks are generally unsuited for toolchain apps like flutter, due to their focus on desktop and unavailability in the cli. But that's of track as i mentioned last time i looked into this Flutter had a requirement of write access in it's root installation folder (or something like that, if memory serves), which would be an issue in /bin or /usr/bin as well as with flatpaks sandbox (idk about AppImage). There might be a solution or suitable workarounds for that, maybe even already. But the usage of snap itself is on of the workarounds as it's "containers" utilize virtual drives in which flutter can write without issue.

**Comment by SchoolGuy** [comment]
That is not a problem for both Appimage and Flatpack.

Flatpacks can in theory just allow direct filesystem access. Using portals would be desired but I don't think that it is strictly required.

Appimage is just a way to ship an application afaik.

Both Appimage and Flatpack have no specific focus on the desktop. Using those with the CLI is easily possible.

**Comment by esdnm** [comment]
Bro, do you guys even care?  (sorry if this sounds rude)

**Comment by maheshmnj** [comment]
@esdnm 
Reopening, Since this issue has already been prioritized, Should the flutter team think that this issue is irrelevant it will be closed. Please leave it open for the flutter team to decide.
